commands/new_function_test.go

```diff
@@ -255,20 +257,18 @@ func Test_duplicateFunctionName(t *testing.T) {
 func writeFunctionYmlFile(functionName string, lang string, t *testing.T) {
 	var stackYaml string
 
-	stackYaml +=
-		`provider:
-  name: faas
-  gateway: ` + defaultGateway + `
-
-functions:
-`
+	stackYaml += fmt.Sprintf(
+		headerYaml,
+		defaultGateway,
+	)
 
-	stackYaml +=
-		`  ` + functionName + `:
-    lang: ` + lang + `
-    handler: ./` + functionName + `
-    image: ` + functionName + `
-`
+	stackYaml += fmt.Sprintf(
```

Comment:
Not overly keen on this refactoring because now I have to scroll to the top of the screen to find what is going to be printed and where.

For this change I'd prefer to keep the existing implementation with the tag appended, or switch to a Golang template.

Alex

Reply:
@alexellis, I prefer being able to use the same templated string in both the `new_function` and `new_function_test`  so that if anything changes in the future, you only have to change code in one place.

The template idea is a great idea. I'd be okay with using that idea. Would you prefer it in a separate file at the root level (`/templates`), or would having in the same place as where I have the YAML constant formats? Just let me know. Thanks!

Reply:
You could create the template in-line as an exported variable.